avm/res/api-management/service/identity-provider/main.bicep

```diff
@@ -14,6 +14,14 @@ param authority string = ''
 @description('Conditional. Client ID of the Application in the external Identity Provider. Required if identity provider is used.')
 param clientId string = ''
 
+@description('Optional. The client library to be used in the developer portal. Only applies to AAD and AAD B2C Identity Provider.')
+@allowed([
+  'ADAL'
+  'MSAL-2'
+  ''
+])
+param clientLibrary string = ''
```

Comment:
```suggestion
param clientLibrary string?
```

Reply:
Good suggestion - but - @tony-box & @anotherRedbeard make sure to validate that the deployment works afterwards. Sometimes, RPs treat `''` & `?` in the same way, sometimes only one or the other is supported.

Also @anotherRedbeard please re-run the `Set-AVMModule` script again after applying the change so that the `main.json` & `readme.md` files are updated